DESIGN.md

```diff
@@ -0,0 +1,183 @@
+    - Start with some predefined curations
+    - If users want to change or overwrite these curations they can
+3. Add index page to list repos
+    - Each repo should have a detailed report (html webapp)
+4. Instructions for program managers and engineers
+    - How to run analysis if you are a program manager
+    - How to run analysis if you are an engineer
+    - How to update curations
+
+## Technical implementation details
+
+The tool will be written as a GitHub action so it can be called in a GitHub workflow
+The action has 2 modes:
+
+### List of repos
+
+Steps in a Github workflow:
+
+1. Weekly scheduled (eg [fairtally-test](https://github.com/jmaassen/fairtally-test/blob/main/.github/workflows/fairtally.yml))
+2. Action fetches repos from RSD
+3. Use ort Docker image from https://hub.docker.com/r/philipssoftware/ort/
+4. Run https://github.com/tortellini-tools/action/blob/rsd-software-vs-ort/ort/batch.sh, replace shell script with Typescript
+5. Create `index-<timestamp>.json` with stats of all repos using Typescript
+6. Create symlink/copy `index-<timestamp>.json` to index-latest.json
+7. To S3 Upload index-latest.json, `index-<timestamp>.json` and for each repo
+    - scan-report-web-app.html,
+    - out.txt
+    - evaluation-result.yml
+8. Vue app index.html which shows index-latest.json in table
+9. Upload files to S3, the S3 bucket can be visted with web browser
+
+Steps 2 .. 8 could be captured in Github Actions like
+
+```yaml
+on:
+    schedule:
+        - cron: '0 0 * * 4' # Every thursday
+
+    # Allows you to run this workflow manually from the Actions tab
+    workflow_dispatch:
+jobs:
+    tortellini:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Get the data dump from the RSD
+              run: curl https://research-software.nl/api/software > software.json
+
+            - name: Extract the list of URLs
+              run: cat software.json | jq -r '[.[].repositoryURLs.github] | flatten | .[]' > urls.txt
+
+            - name: Run ort on urls
+              uses: tortellini-tools/action@v3
+              with:
+                  repositories: urls.txt
+                  output-dir: results
+                  curations: conf/curations.yml
+                  rules: conf/rules.kts
+                  classifications: conf/license-classifications.yml
+            - uses: jakejarvis/s3-sync-action@master
+              with:
+                  args: --acl public-read --follow-symlinks
+              env:
+                  AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
+                  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+                  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+                  AWS_REGION: 'us-west-1' # optional: defaults to us-east-1
+                  SOURCE_DIR: 'results'
+```
+
+### Single repo
+
+Using action for single repo is very similar to multi repo, but works on the currently checkout repository instead of a list of URLs.
```

Comment:
```suggestion
Using the GitHub Action for a single repo is very similar to using it for multiple repositories, but works on the currently checked out repository instead of on a list of URLs.
```